spec/policy/oidc_authentication/oidc_authentication_spec.lua

```diff
@@ -0,0 +1,185 @@
+local _M = require('apicast.policy.oidc_authentication')
+local JWT = require('resty.jwt')
+local rsa = require('fixtures.rsa')
+local oidc_discovery = require('resty.oidc.discovery')
+local OIDC = require('apicast.oauth.oidc')
+local test_backend_client = require 'resty.http_ng.backend.test'
+
+local access_token = setmetatable({
+  header = { typ = 'JWT', alg = 'RS256', kid = 'somekid' },
+  payload = {
+    iss = 'http://example.com/issuer',
+    sub = 'some',
+    aud = 'one',
+    exp = ngx.now() + 3600,
+  },
+}, { __tostring = function(jwt) return JWT:sign(rsa.private, jwt) end })
+
+describe('oidc_authentication policy', function()
+  describe('.new', function()
+    it('works without configuration', function()
+      assert(_M.new())
+    end)
+
+    it('accepts empty configuration', function()
+        assert(_M.new({ }))
+    end)
+
+    it('accepts issuer_endpoint configuration', function()
+      assert(_M.new({ issuer_endpoint = 'http://example.com/path' }))
+    end)
+
+    it('ignores invalid issuer_endpoint configuration', function()
+      assert(_M.new({ issuer_endpoint = 'http: example.com path' }))
+    end)
+
+    local test_backend
+
+    before_each(function()
+      test_backend = test_backend_client.new()
+    end)
+
+    after_each(function()
+      test_backend.verify_no_outstanding_expectations()
+    end)
+
+    it('uses discovery', function()
+      local issuer_endpoint = 'http://example.com'
+
+      stub(oidc_discovery, 'http_backend', test_backend)
+
+      test_backend
+              .expect{ url = issuer_endpoint .. "/.well-known/openid-configuration" }
+              .respond_with{ status = 200, headers = { content_type = 'application/json;charset=UTF-8' },
+                             body = [[
+            {
+              "issuer": "https://example.com/issuer",
+              "jwks_uri": "https://example.com/jwks",
```

Comment:
I think that extracting this one to a local var used in both mocks would nicely show how they're related.